src/gitService.ts

```diff
@@ -935,6 +937,17 @@ export class GitService extends Disposable {
         }
         return user;
     }
+    // TODO: Clear cache when git config changes
+    private _userEmailMapCache: Map<string, string | undefined> = new Map();
+
+    async getCurrentUserEmail(repoPath: string) {
```

Comment:
Instead of adding `getCurrentUserEmail`, let's rename `getCurrentUsername` to `getCurrentUser` and have it return `{ name?: string; email?: string } | undefined`. Then we can share the same cache too

Reply:
Probably would also be best to add a new method in `git.ts`:
```ts
    static async config_getRegex(pattern: string, repoPath?: string) {
        try {
            const data = await gitCommandCore({ cwd: repoPath || '' }, 'config', '--get-regex', pattern);
            return data.trim();
        }
        catch {
            return undefined;
        }
    }
```

That new method can be called from `getCurrentUser` using the pattern `user.(name|email)`.
It will then return output that looks like:
```
> git config --get-regex "user.(name|email)"
user.name Foo Bar
user.email foo@bar.com
```
which could then be easily parsed and returned

Reply:
It seems the returned results in ```config_getRegex``` is got by executing a command in command line and returning its results, but I tried in a test repo, and when the repo's username or email is reset, it may return multiple lines and it will make the judegment rather ambiguous. It seems using ```config_get``` would be a better choice